tests/test_core.py

```diff
@@ -290,6 +290,45 @@ def __test(infile):
         yield (__test, infile)
 
 
+def test_istft_reconstruction():
+    from scipy.signal import bartlett, hann, hamming, blackman, blackmanharris
+
+    def __test(x, n_fft, hop_length, window):
+        S = librosa.core.stft(
+            x, n_fft=n_fft, hop_length=hop_length, window=window)
+        x_reconstructed = librosa.core.istft(
+            S, hop_length=hop_length, window=window)
+
+        L = min(len(x), len(x_reconstructed))
+        x = np.resize(x, L)
+        x_reconstructed = np.resize(x_reconstructed, L)
+
+        # NaN should not happen
+        assert not np.any(np.isnan(x_reconstructed))
```

Comment:
``np.all(np.isfinite(..))`` might be a better choice, in case the normalization causes an underflow to ``+-Inf`` that wouldn't be caught by ``NaN``.  ``isfinite`` should catch both.